test/Python/Poetry/PyProjectSpec.hs

```diff
@@ -0,0 +1,146 @@
+module Python.Poetry.PyProjectSpec (
+  spec,
+) where
+
+import Data.Map qualified as Map
+import Data.Text.IO qualified as TIO
+
+-- import Strategy.Python.Poetry.Common (buildPyProjectGraph)
+import Strategy.Python.Poetry.PyProject (
+  PoetryDependency (..),
+  PyProject (..),
+  PyProjectBuildSystem (..),
+  PyProjectPoetry (..),
+  PyProjectPoetryDetailedVersionDependency (..),
+  PyProjectPoetryGitDependency (..),
+  PyProjectPoetryPathDependency (..),
+  PyProjectPoetryUrlDependency (..),
+  parseConstraintExpr,
+  pyProjectCodec,
+ )
+import Test.Hspec (
+  Expectation,
+  Spec,
+  describe,
+  it,
+  runIO,
+  shouldBe,
+ )
+import Toml qualified
+
+import Data.Text (Text)
+import Data.Void (Void)
+import DepTypes (
+  VerConstraint (
+    CAnd,
+    CCompatible,
+    CEq,
+    CGreater,
+    CGreaterOrEq,
+    CLess,
+    CLessOrEq,
+    CNot,
+    COr
+  ),
+ )
+import Test.Hspec.Megaparsec (shouldParse)
+import Text.Megaparsec (Parsec, parse)
+
+parseMatch :: (Show a, Eq a) => Parsec Void Text a -> Text -> a -> Expectation
+parseMatch parser input expected = parse parser "" input `shouldParse` expected
+
+shouldParseInto :: Text -> VerConstraint -> Expectation
+shouldParseInto = parseMatch parseConstraintExpr
+
+expectedPyProject :: PyProject
+expectedPyProject =
+  PyProject
+    { pyprojectBuildSystem = Just $ PyProjectBuildSystem{buildBackend = "poetry.core.masonry.api"}
+    , pyprojectPoetry =
+        Just $
+          PyProjectPoetry
+            { name = Just "test_name"
+            , version = Just "test_version"
+            , description = Just "test_description"
+            , dependencies =
+                Map.fromList
```

Comment:
Note this gets translated into `1.1` when heading to the backend. Our backend unable to solve version constraint is added as limitation in poetry doc (fyi: we are unable resolve version only when poetry.lock does not exist)